packages/@aws-cdk/assert/lib/assertions/have-output.ts

```diff
@@ -0,0 +1,82 @@
+import {Assertion} from "../assertion";
+import {StackInspector} from "../inspector";
+
+class HaveOutputAssertion extends Assertion<StackInspector> {
+  constructor(private readonly outputName?: string, private readonly exportName?: string, private outputValue?: any) {
+    super();
+  }
+
+  public get description(): string {
+    return `output\
+${this.outputName ? ' with name ' + this.outputName : ''}\
+${this.exportName ? ' with export name ' + this.exportName : ''}\
+${this.outputValue ? ' with value ' + JSON.stringify(this.outputValue) : ''}\
+`;
+  }
+
+  public assertUsing(inspector: StackInspector): boolean {
+    if (!this.outputName && !this.exportName) {
+      throw new Error('At least one of [outputName, exportName] should be defined');
+    }
+    if (!('Outputs' in inspector.value)) {
+      return false;
+    }
+    return (!this.outputName || this.checkOutputName(inspector)) &&
+        (!this.exportName || this.checkExportName(inspector)) &&
+        (!this.outputValue || this.checkOutputValue(inspector));
+  }
+
+  private checkOutputName(inspector: StackInspector): boolean {
+    return !!this.outputName && (this.outputName in inspector.value.Outputs);
+  }
+
+  private checkExportName(inspector: StackInspector): boolean {
+    let outputs;
+    if (this.outputName) {
+      outputs = [inspector.value.Outputs[this.outputName]];
+    } else {
+      outputs = Object.values(inspector.value.Outputs);
+    }
+    const outputWithExport = outputs.find((output: any) => {
+      return output && output.Export && (output.Export.Name === this.exportName);
+    });
+    return !!outputWithExport;
```

Comment:
All this can be expressed as:

```ts
return Object.entries(inspector.value.Outputs)
  // If needed, filter down to only outputs with the required name
  .filter(([name,]) => !this.outputName || this.outputName === name)
  // Check if any of the outputs has the expected export name
  .find(([, output]) => output.Export?.Name === this.exportName) != null;
```

Reply:
The same applies to other elements... It might be able to perform ALL checks at once by using a `filter` sequence, and checking whether the resulting list is empty or not.

Reply:
Thanks for suggestion. I like this approach and used to refactor some of the code. However putting quite a bit of logic into one/two lines might be overwhelming for other maintainers/supporters. I tried to keep balance in refactored code